### PR TITLE
fix error displayed when entering invalid otp

### PR DESCRIPTION
**Pull request by chrono**
## Description
When fetching saml from onelogin after providing an OTP, no errors were handled. One of the expected errors is the OTP not being valid. Now handles error responses from Onelogin and prints them to the user.

Example message returned when entering an invalid OTP:
```Exception: Onelogin Error: '401' 'Failed authentication with this factor'```

## Related Issue
when typing an invalid otp, the error message was:
```AttributeError: 'NoneType' object has no attribute 'saml_response'```

The stack trace associated with this exception:

```
Traceback (most recent call last):
  File "/usr/local/bin/onelogin-aws-login", line 11, in <module>
    load_entry_point('onelogin-aws-cli==0.1.11', 'console_scripts', 'onelogin-aws-login')()
  File "/usr/local/lib/python3.6/site-packages/onelogin_aws_cli/cli.py", line 65, in login
    raise e
  File "/usr/local/lib/python3.6/site-packages/onelogin_aws_cli/cli.py", line 61, in login
    api.save_credentials()
  File "/usr/local/lib/python3.6/site-packages/onelogin_aws_cli/__init__.py", line 139, in save_credentials
    self.assume_role()
  File "/usr/local/lib/python3.6/site-packages/onelogin_aws_cli/__init__.py", line 125, in assume_role
    self.get_role()
  File "/usr/local/lib/python3.6/site-packages/onelogin_aws_cli/__init__.py", line 108, in get_role
    self.get_arns()
  File "/usr/local/lib/python3.6/site-packages/onelogin_aws_cli/__init__.py", line 86, in get_arns
    base64.b64decode(self.saml.saml_response))
AttributeError: 'NoneType' object has no attribute 'saml_response'
```

## How Has This Been Tested?
Applied patch locally and tested by entering valid/invalid otp

### CODE REVIEW

File: onelogin_aws_cli/__init__.py [hunk 1]
```diff
@@ -72,6 +72,11 @@ def get_saml_assertion(self):
                 saml_resp.mfa.state_token,
                 self.mfa.otp
             )
+            if saml_resp is None:
```
Comment (slycoder) [inline onelogin_aws_cli/__init__.py]: Great catch!  How would you feel about pulling this code out into a separate function so there's less duplication with the other None check?

**Comment by drewsonne** [comment]
Is it possible to mock out the `get_saml_assertion` function to add some test coverage for this?

**Comment by jakebolam** [comment]
This keeps tripping developers up

**Comment by chrono** [comment]
CI seems to be failing on flake8, complaining about files that haven't even been touched :/

**Comment by slycoder** [comment]
I'm swamped and can't look into the flake8 issues right now but if anyone else wants to roll that patch into this PR or a separate PR, we'd be glad to have it =).

**Comment by slycoder** [comment]
(Hopefully I can look at it some time next week)

**Comment by slycoder** [comment]
Ok, violations are fixed on master thanks to quick work by @drewsonne .

**Comment by slycoder** [comment]
I think flake8 should be resolved on master; I agree that there should be more test coverage here but we can do that as a followup since there are so many people waiting on this.

**Comment by slycoder** [comment]
Oops, this did in fact break some tests.  I'll patch those up shortly.

**Pull request merged**